Source/WebInspectorUI/UserInterface/Controllers/DeviceSettingsManager.js

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+WI.DeviceSettingsManager = class DeviceSettingsManager extends WI.Object
+{
+    constructor()
+    {
+        super();
+
+        this._deviceUserAgent = null;
+        this._overridenDeviceUserAgent = null;
+        this._overridenDeviceScreenSize = null;
+        this._overridenDeviceSettings = new Map;
+    }
+
+    // Target
+
+    initializeTarget(target)
+    {
+        if (!target.hasDomain("Page"))
+            return
+
+        // COMPATIBILITY (iOS 12.2): Page.overrideUserAgent did not exist.
+        if (target.hasCommand("Page.overrideUserAgent") && this._overridenDeviceUserAgent)
+            target.PageAgent.overrideUserAgent(this._overridenDeviceUserAgent);
+
+        // COMPATIBILITY (iOS 12.2): Page.overrideSetting did not exist.
+        if (target.hasCommand("Page.overrideSetting")) {
+            for (let [setting, value] of this._overridenDeviceSettings)
+                target.PageAgent.overrideSetting(setting, value);
+        }
+
+        const options = {
+            objectGroup: WI.DeviceSettingsManager.ObjectGroup,
```

Comment:
i'm pretty sure you dont need to provide an `objectGroup` as there's no need to keep the return value alive

also, this is only used in this function, so i'd just inline it and then use it as `options.objectGroup` below